tests/integration_tests/modules/test_set_password.py

```diff
@@ -22,6 +22,7 @@
   - name: tom
     # md5 gotomgo
     passwd: "$1$S7$tT1BEDIYrczeryDQJfdPe0"
+    sudo: ALL=(ALL) NOPASSWD:ALL
```

Comment:
I think this actually revealing a bug in the planned implementation: we assume that the default user (with passwordless sudo) will be created as UID 1000. With this user-data, though, `tom` is UID 1000. To maintain the interface with other platforms (we access the instance as `ubuntu` and use `sudo` to gain UID 0), I think `pycloudlib` will need to look up the UID of `self.username` before executing commands.

(As an aside: examining the `cc_users_groups` code, there is no particular guarantee as to what order users will be created in: it depends on dict iteration behaviour. This behaviour is not guaranteed to be order of insertion before Python 3.7 (and _isn't_ order of insertion before CPython 3.6). So we're likely to find that `tom` will not be UID 1000 _some of the time_ on xenial. On 3.6+, as we remove the default user and re-add it, I believe the default user is guaranteed to be the last user created.)

Reply:
Yeah, I honestly thought about that issue, but I assumed that we would always have a non-root user as 1000 to use. But you are totally correct that this assumption is not good and that we should be consistent on using only the ubuntu user to run the commands across different instance types. I will update the pycloudlib code to address that